load-gen/src/main/java/io/smartcat/data/loader/api/RateGenerator.java

```diff
@@ -0,0 +1,6 @@
+package io.smartcat.data.loader.api;
+
+public interface RateGenerator {
+
+    long getRate(long prevInstant, long currentInstant);
```

Comment:
I would suggest adding a variation of getRate method that accepts only the current time instant.